lib/src/date-input/DatePicker.tsx

```diff
@@ -0,0 +1,205 @@
+          onClick={() => handleMonthChange(innerDate.set("month", innerDate.get("month") + 1))}
+        >
+          {rightCaret}
+        </HeaderButton>
+      </PickerHeader>
+      {content === "calendar" && (
+        <Calendar
+          innerDate={innerDate}
+          selectedDate={selectedDate}
+          onInnerDateChange={setInnerDate}
+          onDaySelect={handleDateSelect}
+        />
+      )}
+
+      {content === "yearPicker" && (
+        <YearPicker
+          onYearSelect={(year) => {
+            setInnerDate(innerDate.set("year", year));
+            setContent("calendar");
+          }}
+          selectedDate={selectedDate}
+        />
+      )}
+    </DatePicker>
+  );
+};
+
+const DatePicker = styled.div`
+  padding-top: 16px;
+  background-color: ${(props) => props.theme.dateInput.pickerBackgroundColor};
+  box-shadow: 0 4px 6px -1px rgba(0, 0, 0, 0.1);
+  border: 1px solid #bfbfbf;
+  border-radius: 4px;
+`;
+
+const PickerHeader = styled.div`
+  display: flex;
+  gap: 8px;
+  align-items: center;
+  justify-content: space-between;
+  box-sizing: border-box;
+  height: 40px;
+  padding: 0px 16px;
+`;
+
+const HeaderButton = styled.button`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  width: 24px;
+  height: 24px;
+  padding: 0px;
+  background-color: ${(props) => props.theme.dateInput.pickerMonthArrowsBackgroundColor};
+  font-size: 1.5rem;
+  border-radius: 4px;
+  border: none;
+  cursor: pointer;
+  &:focus {
+    outline: ${(props) => props.theme.dateInput.pickerFocusColor + " solid 2px"};
+  }
+  &:hover {
+    background-color: ${(props) => props.theme.dateInput.pickerHoverDateBackgroundColor};
+  }
+  &:active {
+    background-color: ${(props) => props.theme.dateInput.pickerActiveDateBackgroundColor};
+    color: ${(props) => props.theme.dateInput.pickerActiveDateFontColor};
+  }
+  svg {
+    display: inline-block;
+    flex-shrink: 0;
+    width: 1em;
+    height: 1em;
+    font-size: 1.5rem;
+    user-select: none;
+  }
+`;
+
+const HeaderYearTrigger = styled.button`
+  display: flex;
+  gap: 8px;
+  align-items: center;
+  justify-content: center;
+  height: 40px;
+  width: 172px;
+  color: ${(props) => props.theme.dateInput.pickerMonthFontColor};
+  background-color: transparent;
+  border-radius: 4px;
+  border: none;
+  cursor: pointer;
+  &:hover {
+    background-color: ${(props) => props.theme.dateInput.pickerHoverDateBackgroundColor};
+  }
+  &:focus {
+    outline: ${(props) => props.theme.dateInput.pickerFocusColor + " solid 2px"};
+  }
+  &:active {
+    color: ${(props) => props.theme.dateInput.pickerActiveDateFontColor};
+    background-color: ${(props) => props.theme.dateInput.pickerActiveDateBackgroundColor};
+  }
+`;
+
+const HeaderYearTriggerLabel = styled.span`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  font-family: ${(props) => props.theme.dateInput.pickerFontFamily};
+  font-size: 0.875rem;
+  font-weight: 400;
```

Comment:
Both `font-weight` and `line-height` I think they are unnecessary here (default value and don't do anything, respectively).